src/Evolution/Systems/CurvedScalarWave/Worldtube/SingletonActions/InitializeElementFacesGridCoordinates.hpp

```diff
@@ -0,0 +1,100 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <array>
+#include <cstddef>
+#include <optional>
+#include <unordered_map>
+#include <utility>
+#include <vector>
+
+#include "Domain/Block.hpp"
+#include "Domain/Domain.hpp"
+#include "Domain/ElementMap.hpp"
+#include "Domain/InterfaceLogicalCoordinates.hpp"
+#include "Domain/Structure/CreateInitialMesh.hpp"
+#include "Domain/Structure/ElementId.hpp"
+#include "Domain/Structure/ExcisionSphere.hpp"
+#include "Domain/Structure/InitialElementIds.hpp"
+#include "Domain/Tags.hpp"
+#include "Evolution/DiscontinuousGalerkin/Initialization/QuadratureTag.hpp"
+#include "Evolution/Systems/CurvedScalarWave/Worldtube/Tags.hpp"
+#include "NumericalAlgorithms/Spectral/Mesh.hpp"
+#include "NumericalAlgorithms/Spectral/Spectral.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace CurvedScalarWave::Worldtube::Initialization {
+
+/*!
+ * \brief Initializes a map of the grid coordinates of all faces that abut the
+ * worldtube with corresponding ElementIds on the worldtube singleton.
+ *
+ * \details The worldtube singleton computes an internal solution in the grid
+ * frame and evaluates it at the grid coordinates of each element face abutting
+ * the worldtube each time step. This data is sent to the corresponding element
+ * where it is used to apply pointwise boundary conditions. The
+ * `ElementFacesGridCoordinates` holds a map of all the element ids abutting the
+ * worldtube with the corresonding grid coordinates.
+ *
+ * \warning This currently assumes that initial domain remains the same and
+ * there is no AMR. To support this, the worldtube could send the
+ * coefficients of its internal solution to each element which can evaluate the
+ * it on their current grid in the boundary conditions.
```

Comment:
delete "the"

I think the suggestion here could be: The elements could send their face grid coordinates alongside the data in each timestep, so the worldtube can respond with boundary conditions on these coordinates.

Reply:
it seems a lot easier to just send the coefficients of its internal Taylor series and have each element evaluate it on its own face each time step.